### PR TITLE
Use either of the hotword engine when one is not properly installed

### LINKED ISSUE #215: Make an automated support for using either Snowboy or Pocketsphinx when one is not installed.
**Is your feature request related to a problem? Please describe.**

Currently under `hotword`, we let the user decide if they have Snowboy installed on their device and ask them to use it if they have (due to its much more features). On the other hand if Snowboy is not installed we ask them to go for PocketSphinx and use it.  

https://github.com/fossasia/susi_linux/blob/912948bf34eb43018f71fde44e6ff5ea44eaab2a/main/hotword_engine/__init__.py#L7

Instead of this we can easily do it through the code for the user by checking if the corresponding library is available without the user having to bother which hotword detection to import.

**Describe the solution you'd like**
Make a `try catch ` exception while importing the hotword detection library under https://github.com/fossasia/susi_linux/blob/master/main/hotword_engine/__init__.py

https://github.com/fossasia/susi_linux/blob/912948bf34eb43018f71fde44e6ff5ea44eaab2a/main/hotword_engine/__init__.py#L10

**A clear and concise description of what you want to happen.**

So we can see if we get an `ImportError` while importing snowboy, then we can import the Pocketsphinx and vice-versa. This will not let the user to bother about which library to used without having him/her to check which one is installed. It would simply try importing the libraries and if exception occurs in one it will go for importing the other module.

**Describe alternatives you've considered**
NA

**Additional context**
NA

**Would you like to work on this ?**
I would like to work on it

### PR DESCRIPTION
**Pull request by prateekiiest**
- [x] Delete Local Snowboy folder
- [x] Fix import in snowboy_detector
- [x] Make necessary changes under __init_.py
- [x] Dont copy snowboy local folder
 
Fixes #215 .

We need to decide if this can be a preferable approach

### CODE REVIEW

**Comment by prateekiiest** [comment]
This caan be done once snowboy folder is deleted.
Have sent a PR here #231

**Comment by hongquan** [comment]
@prateekiiest I notice that the local `snowboy` folder contains _susi.pmdl_ file, which seems to be our resource file for `snowboy`. So please find a way to keep it.

**Comment by prateekiiest** [comment]
So we can have the directory structure something like this ?

hotword_engine

- __init__.py	
- hotword_detector.py	
- snowboy_detector.py
- sphinx_detector.py
- susi.pmdl

**Comment by hongquan** [comment]
Can.

**Comment by prateekiiest** [comment]
Yes @stealthanthrax.
I will be able to work on this once the #230 gets merged

**Comment by stealthanthrax** [comment]
@prateekiiest , please incorporate the changes in the [PR](https://github.com/fossasia/susi_linux/pull/231) , in this PR

**Comment by prateekiiest** [comment]
@stealthanthrax I have incorporated #231 changes in this PR

**Comment by hongquan** [comment]
Did you test the new code on your machine?

**Comment by prateekiiest** [comment]
yes @hongquan I have tested it. I had some issues recently regarding setuptools installation while installing for snowboy from the git url. So had to spend some time to fix on that.

**Comment by prateekiiest** [comment]
Had some issues with local installation of snowboy recently. So I need to re-confirm the install.sh script

File: config_generator.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import requests
 from pathlib import Path
 import os
+import imp
```
Comment (hongquan) [inline config_generator.py]: Don't use this module. This one is [deprecated](https://docs.python.org/3/library/imp.html).

File: config_generator.py [hunk 2]
```diff
@@ -62,24 +63,25 @@ def request_hotword_choice():
...
-                print('\nSnowboy set as default Hotword Detection Engine\n')
-            else:
-                config['hotword_engine'] = 'PocketSphinx'
-                print('\nPocketSphinx set as default Hotword Detection Engine\n')
-        else:
-             raise Exception
+        imp.find_module('snowboy')
+        found = True
 
     except Exception:
```
Comment (hongquan) [inline config_generator.py]: Please replace this generic `Exception` to something more specific.

File: main/hotword_engine/__init__.py [hunk 3]
```diff
@@ -7,5 +7,29 @@
...
+except ImportError:
+    pass
+
+try:
+    from .sphinx_detector import PocketSphinxDetector
+    POCKETSPHINX_AVAILABLE = True
+except ImportError:
+    pass
+
+if SNOWBOY_AVAILABLE is True:
```
Comment (prateekiiest) [inline main/hotword_engine/__init__.py]: you might be seeing an old commit. I have already updated it here

**Pull request merged**